client/tests/integration/smartcontracts/executor_with_custom_permission/src/lib.rs

```diff
@@ -199,14 +201,13 @@ fn visit_unregister_domain(
 pub fn migrate(_block_height: u64) -> MigrationResult {
     let accounts = Executor::get_all_accounts_with_can_unregister_domain_permission()?;
 
-    let mut schema = default_permission_schema();
-    schema.remove::<iroha_executor::default::tokens::domain::CanUnregisterDomain>();
-    schema.insert::<token::CanControlDomainLives>();
+    let mut data_model = DataModelBuilder::new();
+    data_model.extend_with_default_permissions();
+    data_model
+        .remove_permission::<iroha_executor::default::permissions::domain::CanUnregisterDomain>();
+    data_model.add_permission::<token::CanControlDomainLives>();
 
-    let (token_ids, schema_str) = schema.serialize();
-    iroha_executor::set_permission_schema(
-        &iroha_executor::data_model::permission::PermissionSchema::new(token_ids, schema_str),
-    );
+    iroha_executor::set_data_model(&data_model.serialize());
```

Comment:
```suggestion
    iroha_executor::set_data_model(&data_model);
```

can't we have api like this?

Reply:
Well, we can, but that would be an unnecessary opinionated API IMO, because `set_data_model` really needs only a borrowed `ExecutorDataModel`. It isn't exactly necessary for developers to rely on the builder, they might want to pass their own `ExecutorDataModel`.

However, this might be better?

```rust
data_model.set(); // invokes `iroha_executor::set_data_model` inside
```

Reply:
Revised builder API, here is before vs after:

```rust
let mut data_model = DataModelBuilder::new();
data_model.extend_with_default_permissions();
data_model
    .remove_permission::<iroha_executor::default::permissions::domain::CanUnregisterDomain>();
data_model.add_permission::<token::CanControlDomainLives>();

iroha_executor::set_data_model(&data_model.serialize());
```

```rust
DataModelBuilder::with_default_permissions()
    .remove_permission::<iroha_executor::default::permissions::domain::CanUnregisterDomain>()
    .add_permission::<token::CanControlDomainLives>()
    .set();

```